### PR TITLE
Use the new ThemeData.platform field rather than global variables

### PR DESCRIPTION
**Pull request by stefanrusek**
This has a number of benefits. First, it switches platforms in a way
that is compatible with the handful of Flutter built-in adaptive widgets
(ie TextField.adaptive()). Second, it removes the global variables, so
the rendered widget tree is based on the state of that widget tree.

### CODE REVIEW

**Comment by aqwert** [comment]
Thanks, looks good as a quick look but will review more closely soon.

File: lib/src/platform_app.dart [hunk 1]
```diff
@@ -260,9 +260,9 @@ class PlatformApp extends PlatformWidgetBase<CupertinoApp, MaterialApp> {
           data?.showSemanticsDebugger ?? showSemanticsDebugger,
       debugShowCheckedModeBanner:
           data?.debugShowCheckedModeBanner ?? debugShowCheckedModeBanner,
-      theme: data?.theme,
+      theme: data?.theme?.copyWith(platform: TargetPlatform.android),
```
Comment (aqwert) [inline lib/src/platform_app.dart]: Fixes the switching of platforms
```suggestion
     theme: data?.theme?.copyWith(platform: TargetPlatform.android) ??
            Theme.of(context).copyWith(platform: TargetPlatform.android),
```

Comment (stefanrusek) [reply lib/src/platform_app.dart]: fixed, thanks

**Comment by stefanrusek** [comment]
@aqwert I added some docs on how platform switching works.

**Comment by aqwert** [comment]
Thanks for your effort on this. Since v 1.9.1 came out I had to update to that to support new widget arguments and not break web since now it is merged into the main flutter framework. Could you get latest and merge in to resolve the conflicts.
After that I will merge into master and release a new version.

**Comment by stefanrusek** [comment]
@aqwert done 👍

File: lib/src/platform.dart [hunk 2]
```diff
@@ -6,53 +6,38 @@
 
 import 'dart:io' show Platform;
```
Comment (cbenhagen) [inline lib/src/platform.dart]: I haven't checked, but it seems that this import is not needed anymore. If it is, can you try to get rid of it? Flutter master will not build web projects with dart:io imports anymore.

Comment (stefanrusek) [reply lib/src/platform.dart]: fixed and verified that it works in browser.

**Pull request merged**